src/trading/NewOrders.sol

```diff
@@ -0,0 +1,338 @@
+        return orders[_orderId].worseOrderId;
+    }
+
+    function getGasPrice(bytes20 _orderId, Trading.TradeTypes _type, IMarket _market, uint8 _outcome) public constant returns (uint256) { 
+        return orders[_orderId].gasPrice;
+    }
+
+    function getBestOrderId(Trading.TradeTypes _type, IMarket _market, uint8 _outcome) public constant returns (bytes20) { 
+        return bestOrder[getBestOrderWorstOrderHash(_market, _outcome, _type)];
+    }
+
+    function getWorstOrderId(Trading.TradeTypes _type, IMarket _market, uint8 _outcome) public constant returns (bytes20) { 
+        return worstOrder[getBestOrderWorstOrderHash(_market, _outcome, _type)];
+    }
+
+    // FIXME: Currently, getOrderIdHash(), getBestOrderWorstOrderHash(), and makeOrder.makeOrder() are creating ripemd160 hash values of type bytes20.  This should probably be changed to create sha3/sha256 hash values of type bytes32.
+    function getOrderIdHash(Trading.TradeTypes _type, IMarket _market, uint256 _fxpAmount, uint256 _fxpPrice, address _sender, uint256 _blockNumber, uint256 _outcome, uint256 _fxpMoneyEscrowed, uint256 _fxpSharesEscrowed) public constant returns (bytes20) {
+        return ripemd160(_type, _market, _fxpAmount, _fxpPrice, _sender, _blockNumber, _outcome, _fxpMoneyEscrowed, _fxpSharesEscrowed);
+    }
+
+    function getBlockNumber() public constant returns (uint256) {
+        return block.number;
+    }
+
+    function isBetterPrice(Trading.TradeTypes _type, IMarket _market, uint8 _outcome, uint256 _fxpPrice, bytes20 _orderId) public constant returns (bool) { 
+        if (_type == Trading.TradeTypes.Bid) {
+            return (_fxpPrice > orders[_orderId].fxpPrice);
+        } else {
+            return (_fxpPrice < orders[_orderId].fxpPrice);
+        }
+    }
+
+    function isWorsePrice(Trading.TradeTypes _type, IMarket _market, uint8 _outcome, uint256 _fxpPrice, bytes20 _orderId) public constant returns (bool) { 
+        if (_type == Trading.TradeTypes.Bid) { 
+            return (_fxpPrice < orders[_orderId].fxpPrice);
+        } else {
+            return (_fxpPrice > orders[_orderId].fxpPrice);
+        }
+    }
+
+    function assertIsNotBetterPrice(Trading.TradeTypes _type, IMarket _market, uint8 _outcome, uint256 _fxpPrice, bytes20 _betterOrderId) public constant returns (bool) { 
+        require(!isBetterPrice(_type, _market, _outcome, _fxpPrice, _betterOrderId));
+        return true;
+    }
+
+    function assertIsNotWorsePrice(Trading.TradeTypes _type, IMarket _market, uint8 _outcome, uint256 _fxpPrice, bytes20 _worseOrderId) public returns (bool) { 
+        require(!isWorsePrice(_type, _market, _outcome, _fxpPrice, _worseOrderId));
+        return true;
+    }
+
+    function insertOrderIntoList(bytes20 _orderId, Trading.TradeTypes _type, IMarket _market, uint8 _outcome, uint256 _fxpPrice, bytes20 _betterOrderId, bytes20 _worseOrderId) internal onlyWhitelistedCallers returns (bool) { 
```

Comment:
Functions that are `internal` don't need `onlyWhitelistCallers`.  Serpent didn't have `private` or `internal` functions so we ended up flagging such methods as `onlyWhitelistedCallers`.

In general, any method that had `assertPrivate` or `onlyWhitelistedCallers` and was never called outside of the contract should be changed to `private`.  `internal` should only be used if you are leveraging inheritance and need to expose something to a derived contract (pretty rare).  If it had `onlyWhitelistedCallers` and was called from other Augur contracts, then it should be public with the `onlyWhitelistedCallers` modifier.